src/anndata/_core/anndata.py

```diff
@@ -1773,8 +1778,11 @@ def concatenate(
 
         # Backwards compat (some of this could be more efficient)
         # obs used to always be an outer join
+        sprase_class = sparse.csr_matrix
+        if any(isinstance(a.X, SpArray) for a in all_adatas):
```

Comment:
Should we remove this for codecov to be happy since the function is already deprecated?

Reply:
I don't think this was tested for anyway